crates/evm/src/query.rs

```diff
@@ -36,14 +36,30 @@ use crate::rpc_helpers::*;
 use crate::{BloomFilter, EthResult, Evm, EvmChainConfig, FilterBlockOption, FilterError};
 
 // Gas per transaction not creating a contract.
-const MIN_TRANSACTION_GAS: u64 = 21_000u64;
+pub(crate) const MIN_TRANSACTION_GAS: u64 = 21_000u64;
 
 /// https://github.com/paradigmxyz/reth/pull/7133/files
 /// Allowed error ratio for gas estimation
 /// Taken from Geth's implementation in order to pass the hive tests
 /// <https://github.com/ethereum/go-ethereum/blob/a5a4fa7032bb248f5a7c40f4e8df2b131c4186a4/internal/ethapi/api.go#L56>
 const ESTIMATE_GAS_ERROR_RATIO: f64 = 0.015;
 
+/// The result of gas estimation.
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub(crate) struct EstimatedGas {
+    /// Evm gas used.
+    gas: U64,
+    /// L1 fee.
+    l1_fee: U256,
+}
+
+impl EstimatedGas {
```

Comment:
this is wrong

gas field is gas used, not gas fee.

l1 fee is diff size * l1 fee rate

we want this to be  (l1_fee / base_fee) + gas 

so that when a wallet etc. tries to estimate if its tx goes through, 

it can do the following:

1. call eth_estimateGas
2. call eth_maxFeePerGas
3. multiply returned values
4. check balance